Остапович Дмитрий

WaterRaven opened the pull request.

WaterRaven committed:
Created Body

Создан файл класса Body, написан конструктор

WaterRaven committed:
move method added

WaterRaven committed:
destroy and merge methods added

WaterRaven commented:
Пока только начало, самое интересное будет позже

WaterRaven committed:
Код отформатирован

WaterRaven committed:
Changed file name + some funcs added

Добавлены функции для вычисления ускорения и функция gravitate

WaterRaven committed:
Merge branch 'vongostev:main' into main

WaterRaven committed:
Commentaries added

Добавлены комментарии функций

WaterRaven committed:
Gravitation class created

Добавлен класс Гравитации - движка, определяющего работу гравитации. Все функции и глобальные переменные перемещены в него. Добавлена структура логов траекторий

WaterRaven committed:
read_trajectory added

Добавлен генератор точек траектории по файлу

WaterRaven committed:
Tests added

Начал добавлять юнит-тесты. Создание и движение тела

vongostev commented:
Добрый вечер!
Все выглядит супер, но нет тестов, непонятно, работает ли код вообще. Приложите ноутбук с результатами тестов.
Почему метод `destroy` помещает тело в начало координат?
Все вычисления типа `attractor_mass * self.G * distance / np.linalg.norm(distance)**3` я бы вынес в отдельную ускоренную функцию. В Вашей архитектуре все ускорения можно считать параллельно с векторизацией, будет значительно эффективнее.
Вот такая строчка `for attractor in [a for a in self.bodies if a != body]:` будет работать медленнее, чем `for attractor in self.bodies: if attractor != body`.

WaterRaven commented:
Над тестами сейчас работаю. Поправки также сейчас учту

WaterRaven commented:
Перемещение в центр координат не имеет практической цели, т.к. тело всё равно перестаёт двигаться и воздействовать на прочие тела. На анимации оно также не будет отображаться. Сделано это лишь для облегчения ручного чтения файла с логами

WaterRaven committed:
read func tested (and fixed)

Также вынесены из классов функция чтения траектории и функция вычисления ускорения. Последняя ускорена нумбой

WaterRaven committed:
fixed tests, added test for calculate_acceleration

Также уничтожение больше не перемещает в начало координат

WaterRaven committed:
minor changes

vongostev commented:
Доброе утро!
Все отлично. Добавьте только еще графический вывод результатов, и все.

WaterRaven committed:
accelerated calculations, added animation (NOT TESTED YET)

Также исправлены некоторые нестыковки с конвертацией типов int в float

WaterRaven committed:
I tried my best

The animation in python is f***ed as hell

vongostev commented:
Добрый вечер! Все отлично, но не мучайтесь с анимацией и выведите её в блокнот юпитер, это займет 10 строчек.

WaterRaven committed:
I won

Я победил анимацию, она работает. Пока не может рисовать след, но сделаю и это

WaterRaven committed:
Animation is DONE

WaterRaven committed:
radius dismissed

Теперь за радиус принимаем кубический корень из массы

vongostev commented:
Анимация так и не работает, все остальное отлично.

WaterRaven commented:
У меня работает...
